rma_sale/static/src/js/rma_portal_form.js

```diff
@@ -4,42 +4,122 @@
 odoo.define("rma_sale.animation", function(require) {
     "use strict";
 
-    var sAnimation = require("website.content.snippets.animation");
+    const publicWidget = require("web.public.widget");
 
-    // In the customer portal when a RMA operation is selected show the comments
-    // selector so the user doesn't miss the chance to add his comments
-    sAnimation.registry.rma_operation_portal = sAnimation.Class.extend({
-        selector: ".rma-operation",
+    /**
+     * Adds some machinery to the customer portal RMA form:
+     *
+     * - Avoid submitting the form if no qty or operation is reported.
+     * - Show automatically the observations field when the operation is selected
+     *   and hide it back with no operation selected.
+     */
+    publicWidget.registry.PortalRmaSale = publicWidget.Widget.extend({
+        selector: "#form-request-rma",
+        events: {
+            "change .rma-operation": "_onChangeOperationId",
+            "change #delivery-rma-qty input": "_onChangeQty",
+        },
+
+        /**
+         * @override
+         */
         start: function() {
-            this.id = this.el.name.replace("-operation_id", "");
-            this.$comment = $("#comment-" + this.id);
-            this.$comment_input = $("[name='" + this.id + "-description']");
+            const ids = $("[name*='-operation_id']")
```

Comment:
plublicWidget extends web.Widget which defines the function "$" which calls "this.$el.find". This ensures that the widget can work together with other widget instances and, in general, not collide if other widgets identify an element in the same way.
```suggestion
            const ids = this.$("[name*='-operation_id']")
```